correlation_context/HTTP_HEADER_FORMAT_RATIONALE.md

```diff
@@ -1,74 +1,72 @@
-- Should be appende-able (comma-separated) https://tools.ietf.org/html/rfc7230#page-24 so nodes 
-can add context properties without parsing an existing headers.
-- It is expected that the typical name will be a single word in latin and the value will be a 
-short string in latin or a derivative of an url.
+- It should be human-readable. Cryptic headers would hide the fact of potential information disclosure.
+- It should be append-able (comma-separated) https://tools.ietf.org/html/rfc7230#page-24 so nodes
+can add context properties without parsing existing headers.
+- Keys are a single word in ASCII, and values should be a short string in UTF-8 or a derivative of an URL.
 
 ## Why a single header?
 
-Another option is to use prefixed headers, such as `trace-context-X` where `X` is a propagated 
-field name. This could reduce the size of the data, particularly in http/2 where header 
+Another option would be to use prefixed headers, such as `trace-context-X`, where `X` is a propagated
+field name. That could reduce the size of the data, particularly in http/2, where header
 compression can apply.
 
-Generally speaking `Correlation-Context` header may be split into multiple headers and 
-compression may be at the same ballpark as repeating values will be converted into a single value
-in HPAC's dynamic collection. That's said there were no profiling made to make this decision.
+Generally speaking, a `Correlation-Context` header may be split into multiple headers, and
+compression may be at the same ballpark as repeating values are converted into a single value
+in HPAC's dynamic collection. That said, there was no profiling made to make this decision.
 
 The approach with multiple headers has the following problems:
-- Name values limitation is much more pressing when the context name is used a part of a header 
+- Name values limitation is much more pressing when the context name is used a part of a header
 name.
-- The comma-separated format similar to the proposed still needs to be supported in every 
-individual header. This makes parsing harder.
-- Single header is easier to configure for tracing by many app servers.
+- The comma-separated format similar to the proposed still needs to be supported in every individual header which makes parsing harder.
+- A single header is more comfortable to configure for tracing by many app servers.
 
 ## Why not Vary-style?
 
-The [Vary](https://tools.ietf.org/html/rfc7231#section-7.1.4) approach is another alternative, 
+The [Vary](https://tools.ietf.org/html/rfc7231#section-7.1.4) approach is another alternative,
 which could be used to accomplish the same. For example, `Correlation-Context: x-b3-parentid;
-ttl=1` could tell the propagation to look at and forward the parent ID header, but only to the 
-next hop. This has an advantage of http header compression (hpack) and also weave-in with legacy 
+ttl=1` could tell the propagation to look at and forward the parent ID header, but only to the
```

Comment:
this part makes even less sense - the interpretation of ttl property is completely unrelated to the discussion of Vary header format.

Reply:
https://github.com/w3c/correlation-context/issues/26